src/js/AppPermissions.js

```diff
@@ -0,0 +1,97 @@
+import React from 'react';
+import { connect } from 'react-redux'
+
+import AppHeader from './containers/Header';
+import sdlController from './Controllers/SDLController';
+import { Link } from 'react-router-dom';
+import store from './store';
+import { closePermissionsView, clearAppAwaitingPermissions } from './actions'
+
+const mapStateToProps = (state) => {
+    var theme = state.theme
+    var editingAppID = state.system.editingPermissionsAppId;
+    var allowedFunctions = state.system.allowedFunctions;
+    var pendingActivation = state.system.permissionsAppAwaitingActivation;
+
+    return {
+        theme: theme,
+        editingAppID: editingAppID,
+        allowedFunctions: allowedFunctions,
+        activeLayout: state.ui[editingAppID]?.displayLayout,
+        appOptions: state.appList,
+        pendingActivation: pendingActivation
+    };
+}
+
+class Permissions extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.toggleFunction = this.toggleFunction.bind(this);
+        this.savePermissions = this.savePermissions.bind(this);
+
+        this.state = {
+            allowedFunctions: props.allowedFunctions
+        }
+    }
+
+    toggleFunction(name) {
+        var functions = [ ...this.state.allowedFunctions ]
+        for (var f of functions) {
+            if (f.name === name) {
+                f.allowed = !f.allowed;
+                this.setState({ allowedFunctions: functions });
+                return;
+            }
+        }
+    }
+
+    savePermissions() {
+        var allowedFunctions = this.state.allowedFunctions.map((f) => ({
+            id: f.id, name: f.name, allowed: f.allowed
+        }));
+        sdlController.onAppPermissionConsent(this.props.editingAppID, allowedFunctions);
+
+        if (this.props.pendingActivation === this.props.editingAppID) {
+            sdlController.onAppActivated(this.props.editingAppID);
```

Comment:
Remove this call to onAppActivated. SDL Core sends an activate app request to hmi after data is consented